File: Composicion-Agregacion/Ejercicio1/Casa.java
package Ejercicio1;

import java.util.ArrayList;

public class Casa {
    public String direccion;
    public ArrayList<Habitacion> habitaciones;

    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new ArrayList<>();
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public void agregarHabitacion(Habitacion habitacion) {
        habitaciones.add(habitacion);
    }

    public void mostrarCasa() {
        System.out.println("Direccion de la casa: " + direccion);
        System.out.println("Habitaciones:");
        for (Habitacion h : habitaciones) {
            h.mostrarInfo();
        }
    }
     public static void main(String[] args) {
        // b) Crea una casa y agrega varias habitaciones
        Casa casa = new Casa("Calle Heroes de Puno");

        Habitacion habitacion1 = new Habitacion("Sala", 25);
        Habitacion habitacion2 = new Habitacion("Cocina", 12);
        Habitacion habitacion3 = new Habitacion("Dormitorio", 20);
        Habitacion habitacion4 = new Habitacion("Baño", 14);

        casa.agregarHabitacion(habitacion1);
        casa.agregarHabitacion(habitacion2);
        casa.agregarHabitacion(habitacion3);
        casa.agregarHabitacion(habitacion4);
        // c) Muestra la información de la casa y sus habitaciones
        casa.mostrarCasa();
    }   
}

File: Composicion-Agregacion/Ejercicio1/Casa.py
# a) Implementa las clases con sus constructores, getters y setters
class Habitacion:
    def __init__(self, nombre, tamano):
        self.nombre = nombre
        self.tamano = tamano

    def get_nombre(self):
        return self.nombre

    def set_nombre(self, nombre):
        self.nombre = nombre

    def get_tamano(self):
        return self.tamano

    def set_tamano(self, tamano):
        self.tamano = tamano

    def mostrar_info(self):
        print(f"Habitación: {self.nombre}, Tamaño: {self.tamano} m²")

class Casa:
    def __init__(self, direccion):
        self.direccion = direccion
        self.habitaciones = []

    def get_direccion(self):
        return self.direccion

    def set_direccion(self, direccion):
        self.direccion = direccion

    def agregar_habitacion(self, habitacion):
        self.habitaciones.append(habitacion)

    def mostrar_casa(self):
        print(f"Dirección de la casa: {self.direccion}")
        print("Habitaciones:")
        for habitacion in self.habitaciones:
            habitacion.mostrar_info()

# b) Crea una casa y agrega varias habitaciones

casa = Casa("Calle Heroes de Puno")

habitacion1 = Habitacion("Sala", 25)
habitacion2 = Habitacion("Cocina", 12)
habitacion3 = Habitacion("Dormitorio", 20)
habitacion4 = Habitacion("Baño", 14)

casa.agregar_habitacion(habitacion1)
casa.agregar_habitacion(habitacion2)
casa.agregar_habitacion(habitacion3)
casa.agregar_habitacion(habitacion4)

# c) Muestra la información de la casa y sus habitaciones

casa.mostrar_casa()